pywbem_mock/_inmemoryrepository.py

```diff
@@ -0,0 +1,264 @@
+                return self._data[name].copy()
+            return self._data[name]
+        else:
+            raise KeyError('{} not in {} object store'
+                           .format(name, self._cim_object_type))
+
+    def create(self, name, cim_object):
+        assert isinstance(cim_object, self._cim_object_type)
+
+        if name in self._data:
+            raise ValueError('{} already in {} object store'
+                             .format(name, self._cim_object_type))
+
+        self._data[name] = cim_object.copy()
+
+    def update(self, name, cim_object):
+        assert isinstance(cim_object, self._cim_object_type)
+
+        if name not in self._data:
+            raise KeyError('{} not in {} object store'
+                           .format(name, self._cim_object_type))
+
+        # Replace the existing object with a copy of the input object
+        self._data[name] = cim_object.copy()
+
+    def delete(self, name):
+        if name in self._data:
+            del self._data[name]
+        else:
+            raise KeyError('{} not in {} object store'
+                           .format(name, self._cim_object_type))
+
+    def iter_names(self):
+        return six.iterkeys(self._data)
+
+    def iter_values(self, copy=True):
+        for value in six.itervalues(self._data):
+            if copy:
+                yield(value.copy())
+            else:
+                yield(value)
+
+    def len(self):
+        return len(self._data)
+
+
+class InMemoryRepository(BaseRepository):
+    """
+    A CIM repository that maintains its data in memory.
+
+    Documentation for the methods and properties inherited from
+    ~pywbem_mock:`BaseObjectStore` is also inherited in the pywbem
+    documentation. Therefore the methods in this class have no
+    documentation string.
+    """
+    def __init__(self, initial_namespace=None):
+        """
+        Initialize the InMemoryRepository.
+
+        Parameters:
+
+          initial_namespace:(:term:`string` or None):
+            Optional initial namespace that will be added to
+            the CIM repository.
+        """
+
+        # Create the in memory repository.  This defines the top level
+        # NocaseDict which defines the namespaces. The keys of this
+        # dictionary are namespace names and the values are the dictionaries
+        # defining the CIM classes, CIM instances, and CIM qualifier
+        # declarations where the keys are "classes", "instance", and
+        # "qualifiers" and the value for each is an instance of the
+        # class InMemoryObjectStore
+        self._repository = NocaseDict()
+
+        # If an initial namespace is defined, add it to the repository
+        if initial_namespace:
+            self.add_namespace(initial_namespace)
+
+    def print_repository(self, dest=None, ):
+        """
+        Display the items in the repository. This displays information on
+        the items in the data base and is only a diagnostic tool.
```

Comment:
I suggest to change the first sentence to: `Print the CIM repository to a destination".

Reply:
DONE